### PR TITLE
Polar coordinate bottom alignment: updated maximum radius

### PR DESCRIPTION
**Pull request by MrRamka**
image

### CODE REVIEW

File: src/core/prototypes/plot_segments/region_2d/utils.ts [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      if (angleDelta <= 90 - startAngle) {
+        //endAngle - 1 quadrant => move left bottom corner
+        cx = x1;
+        cy = y1;
+        radialRatio = 2;
+      } else if (angleDelta <= 180 - startAngle) {
+        //endAngle - 4 quadrant => move left
+        cx = x1;
+        cy = (y2 + y1) / 2;
+        // radialRatio = 2;
```
Comment (jrthompson33) [inline src/core/prototypes/plot_segments/region_2d/utils.ts:51]: Remove commented out code.

File: src/core/prototypes/plot_segments/region_2d/utils.ts [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+        isHalf = true;
+        side = SIDE.VERTICAL;
+      } else if (angleDelta + startAngle > 180) {
+        //endAngle - 3 and 4 quadrants
+        cx = (x2 + x1) / 2;
+        cy = (y2 + y1) / 2;
```
Comment (jrthompson33) [inline src/core/prototypes/plot_segments/region_2d/utils.ts:57]: Should `isHalf` flag be `true` for this condition?

File: src/core/prototypes/plot_segments/region_2d/utils.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        //endAngle - 1 quadrant => move right
+        cx = x2;
+        cy = (y2 + y1) / 2;
+        isHalf = true;
+        side = SIDE.VERTICAL;
+        isRightHalf = true;
+      } else if (angleDelta + startAngle > 360) {
+        //endAngle - 2 and 1 quadrants
+        cx = (x2 + x1) / 2;
+        cy = (y2 + y1) / 2;
```
Comment (jrthompson33) [inline src/core/prototypes/plot_segments/region_2d/utils.ts:95]: Should `isHalf` flag be `true` for this condition?

File: src/core/prototypes/plot_segments/region_2d/utils.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+  props: PolarProperties,
+  attrs: PolarAttributes,
+  center: CenterType,
+  startPoint: number,
+  endPoint: number
+) {
+  const width = Math.abs(attrs.x2 - attrs.x1);
+  const height = Math.abs(attrs.y2 - attrs.y1);
+  if (props.autoAlignment) {
+    if (center.isHalf) {
```
Comment (jrthompson33) [inline src/core/prototypes/plot_segments/region_2d/utils.ts:246]: Could this be:
 `if(props.autoAlignment && center.isHalf) { ` 
 ... 
 ` } else { 
   attrs.radial1 = (props.innerRatio * (endPoint - startPoint)) / 2;
   attrs.radial2 = (props.outerRatio * (endPoint - startPoint)) / 2;
}`

**Review by jrthompson33 — COMMENTED** [review]
PR looks good - it definitely makes sense to move these functions into util class.

Please see my minor comments in the review. There is a few remarks about general code clean-up (e.g., removing commented out code and simplifying if/else blocks). My other comments are more geared towards understanding what the different output parameters of `getCenterByAngle` are being used for.

File: src/core/prototypes/plot_segments/region_2d/utils.ts (hunk 2 above)
Comment (MrRamka) [reply src/core/prototypes/plot_segments/region_2d/utils.ts:57]: For this case `isHalf` flag should be `false`. The chart takes more than 180 degrees.

File: src/core/prototypes/plot_segments/region_2d/utils.ts (hunk 4 above)
Comment (MrRamka) [reply src/core/prototypes/plot_segments/region_2d/utils.ts:246]: Fixed

File: src/core/prototypes/plot_segments/region_2d/utils.ts (hunk 3 above)
Comment (MrRamka) [reply src/core/prototypes/plot_segments/region_2d/utils.ts:95]: For this case `isHalf` flag should be `false`. The chart takes more than 180 degrees.

File: src/core/prototypes/plot_segments/region_2d/utils.ts (hunk 1 above)
Comment (MrRamka) [reply src/core/prototypes/plot_segments/region_2d/utils.ts:51]: Removed

**Comment by MrRamka** [comment]
Thank you for your comments.

**Comment by jrthompson33** [comment]
Thanks for making the minor changes. I've approved this review.

**Review by jrthompson33 — APPROVED** [review]
Changes to the PR improved the code. This PR has been approved.

**Review by bongshin — APPROVED** [review]

**Pull request merged**